### PR TITLE
mosip 20660 Ability to update address and contact information as part…

### PR DESCRIPTION
**Pull request by Shilpa-Manjunath**
… of LOST UIN

### CODE REVIEW

File: registration-processor/core-processor/registration-processor-uin-generator-stage/src/main/java/io/mosip/registration/processor/stages/uingenerator/stage/UinGeneratorStage.java [hunk 1]
```diff
@@ -153,6 +153,9 @@ public class UinGeneratorStage extends MosipVerticleAPIManager {
 	/** After this time intervel, message should be considered as expired (In seconds). */
 	@Value("${mosip.regproc.uin.generator.message.expiry-time-limit}")
 	private Long messageExpiryTimeLimit;
+	
+	@Value("${uingenerator.lost.packet.update.information}")
```
Comment (MonobikashDas) [inline registration-processor/core-processor/registration-processor-uin-generator-stage/src/main/java/io/mosip/registration/processor/stages/uingenerator/stage/UinGeneratorStage.java]: change it to
uingenerator.lost.packet.allowed.update.fields=

File: registration-processor/core-processor/registration-processor-uin-generator-stage/src/main/java/io/mosip/registration/processor/stages/uingenerator/stage/UinGeneratorStage.java [hunk 2]
```diff
@@ -1020,7 +1024,13 @@ private IdResponseDTO linkRegIdWrtUin(String lostPacketRegId, String matchedRegI
 			identityObject.put(UINConstants.UIN, uin);
 			String schemaVersion = packetManagerService.getFieldByMappingJsonKey(lostPacketRegId, MappingJsonConstants.IDSCHEMA_VERSION, process, ProviderStageName.UIN_GENERATOR);
 			identityObject.put(idschemaversion, convertIdschemaToDouble ? Double.valueOf(schemaVersion) : schemaVersion);
-
+			String[] upd= updateInfo.split(",");
```
Comment (MonobikashDas) [inline registration-processor/core-processor/registration-processor-uin-generator-stage/src/main/java/io/mosip/registration/processor/stages/uingenerator/stage/UinGeneratorStage.java]: make default value for updateInfo as null. If no update is allowed then skip this code
also rename the method name -
linkRegIdWrtUin => lostAndUpdateUin()

File: registration-processor/core-processor/registration-processor-uin-generator-stage/src/main/resources/bootstrap.properties [hunk 3]
```diff
@@ -21,3 +23,6 @@ mosip.regproc.eventbus.kafka.commit.type=${mosip.regproc.uin.generator.eventbus.
 mosip.regproc.eventbus.kafka.max.poll.records=${mosip.regproc.uin.generator.eventbus.kafka.max.poll.records}
 mosip.regproc.eventbus.kafka.poll.frequency=${mosip.regproc.uin.generator.eventbus.kafka.poll.frequency}
 mosip.regproc.eventbus.kafka.group.id=${mosip.regproc.uin.generator.eventbus.kafka.group.id}
+
+
+uingenerator.lost.packet.update.information=phone,email
```
Comment (MonobikashDas) [inline registration-processor/core-processor/registration-processor-uin-generator-stage/src/main/resources/bootstrap.properties]: dont add phone email here. Country might call phone as mobileNo in identity schema beacuse of which our code will fail. So dont assume any default value wherever identity schema is involved.

**Review by MonobikashDas — CHANGES_REQUESTED** [review]

File: registration-processor/core-processor/registration-processor-uin-generator-stage/src/main/java/io/mosip/registration/processor/stages/uingenerator/stage/UinGeneratorStage.java (hunk 1 above)
Comment (Shilpa-Manjunath) [reply registration-processor/core-processor/registration-processor-uin-generator-stage/src/main/java/io/mosip/registration/processor/stages/uingenerator/stage/UinGeneratorStage.java]: changed the property

File: registration-processor/core-processor/registration-processor-uin-generator-stage/src/main/java/io/mosip/registration/processor/stages/uingenerator/stage/UinGeneratorStage.java (hunk 2 above)
Comment (Shilpa-Manjunath) [reply registration-processor/core-processor/registration-processor-uin-generator-stage/src/main/java/io/mosip/registration/processor/stages/uingenerator/stage/UinGeneratorStage.java]: done

File: registration-processor/core-processor/registration-processor-uin-generator-stage/src/main/resources/bootstrap.properties (hunk 3 above)
Comment (Shilpa-Manjunath) [reply registration-processor/core-processor/registration-processor-uin-generator-stage/src/main/resources/bootstrap.properties]: ok done

**Review by MonobikashDas — APPROVED** [review]

**Pull request merged**